samples/contrib/azure-samples/databricks-pipelines/compile_upload_run.sh

```diff
@@ -0,0 +1,99 @@
+#! /bin/bash
+
+#
+# Argument parsing and validation
+#
+KUBEFLOW_HOST=""
+PIPELINE_NAME=""
+PIPELINE_PARAMS="[]"
+
+function show_usage() {
+    echo "compile_upload_run.sh"
+    echo
+    echo -e "\t--kubeflow-host\t(Required) Kubeflow server address (e.g. http://localhost:8080)"
+    echo -e "\t--pipeline-name\t(Required) Name of the pipeline (e.g. databricks_operator_run_pipeline)"
+    echo -e "\t--pipeline-name\t(Optional) Pipeline parameters (e.g. '[{\"name\":\"run_name\",\"value\":\"test-run\"},{\"name\":\"parameter\",\"value\":\"10\"}]')"
+}
+
+while [[ $# -gt 0 ]]
+do
+    case "$1" in 
+        --kubeflow-host)
+            KUBEFLOW_HOST="$2"
+            shift 2
+            ;;
+        --pipeline-name)
+            PIPELINE_NAME="$2"
+            shift 2
+            ;;
+        --pipeline-params)
+            PIPELINE_PARAMS="$2"
+            shift 2
+            ;;
+        *)
+            echo "Unexpected '$1'"
+            show_usage
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z $KUBEFLOW_HOST ]; then
+    echo "kubeflow-host not specified"
+    echo
+    show_usage
+    exit 1
+fi
+
+if [ -z $PIPELINE_NAME ]; then
+    echo "pipeline-name not specified"
+    echo
+    show_usage
+    exit 1
+fi
+
+#
+# Main script start
+#
+
+echo "Compiling pipeline '$PIPELINE_NAME.py'..."
+# To test with the local package: pip install --upgrade ../kfp-azure-databricks
+pip install -e "git+https://github.com/kubeflow/pipelines#egg=kfp-azure-databricks&subdirectory=samples/contrib/azure-samples/kfp-azure-databricks" --upgrade
+python ./$PIPELINE_NAME.py
+PIPELINE_FILE="$PIPELINE_NAME.py.tar.gz"
+
+echo "Uploading pipeline '$PIPELINE_FILE' to '$KUBEFLOW_HOST'..."
+UPLOAD_RESPONSE=`curl -s -X POST -F uploadfile=@$PIPELINE_FILE $KUBEFLOW_HOST/pipeline/apis/v1beta1/pipelines/upload?name=$PIPELINE_NAME`
```

Comment:
This api is subject to change in future. Would you consider write something using kfp's python client + [fire](https://github.com/google/python-fire) to launch the pipeline?

Reply:
I'll check this out.

Reply:
I changed the bash script to a Python CLI that uses kfp + fire to do the same thing.

Reply:
Super. Thanks!